### DISCUSSION TITLE
SCSS linter, formatter

### DISCUSSION THREAD

**Discussion by JosepSalvat** [biomejs/biome#3441] [Ideas]
Good morning, 
As far as I know, BiomeJS have support CSS, is it possible to make it work for SCSS also?
We've SCSS Module (like CSS Modules, but with SCSS).
Thanks!

**Comment by ematipico** [comment]
The core team won't focus on this for this year, but you're free to collect interest and maybe evaluate to contribute it

**Reply by YanPes** [reply]
Is there a plan to add it this year?

**Reply by ematipico** [reply]
https://biomejs.dev/blog/roadmap-2025/

**Reply by SalahAdDin** [reply]
@YanPes @ematipico  No plans over Biome.

**Reply by ematipico** [reply]
Why did you tag me? 😅

**Reply by SalahAdDin** [reply]
> https://biomejs.dev/blog/roadmap-2025/

No plans about SCSS.

**Comment by pooya-badiee** [comment]
Would be Great to have, makes us able to delete prettier

**Comment by Danielku15** [comment]
Would it be an option to enable a "best effort parsing" for SCSS (for the VS Code extension)? 

I realized that in many files I use primarily syntax which has landed in standard CSS (CSS modules, CSS nesting, CSS variables). I still use SCSS as primary file format and in some exceptions I still use SCSS features. In most cases Biome should be able to handle my "SCSS" files as they only contain pure "CSS" code.  

But when attempting to format I get a 
![image](https://github.com/user-attachments/assets/6656a463-b6af-40a2-aee7-a6386e2c9474)

**Reply by pooya-badiee** [reply]
I have the same experience. my SCSS looks just like  modern CSS

**Reply by SalahAdDin** [reply]
I also have issues with HTML, so, It seems We will have to comeback to Prettier.

**Reply by muuvmuuv** [reply]
I am also just using the nice nesting from SCSS but no other SCSS features :/

**Reply by ematipico** [reply]
I am a bit afraid of the knockout effect it  could have. It's a nice idea, but as a maintainer I'm afraid that a lot of people will misunderstand what it is, and will start asking for more, or will file issues regardless of what we write in the documentation.

If Biome starts supporting SCSS files, where's the line "this is supported, and this isn't?".

**Reply by artkub** [reply]
I understand your concerns and thank you for your great work on Biome.
Would you consider adding an "extensions" or "matchFiles" option to the CSS section in the config file ?
```
{
  "css": {
    "extensions": [ ".css", ".css.tmpl", ".scss", ".less" ],
    // or
    "matchFiles": [ "src/**/*.scss" ],
  }
}
```
Or something similar that would allow users to define which files should be considered as CSS files. I believe that this would satisfy the use cases discussed above and would avoid any confusion about SCSS being an officially supported file format.

**Reply by SalahAdDin** [reply]
> I am a bit afraid of the knockout effect it  could have. It's a nice idea, but as a maintainer I'm afraid that a lot of people will misunderstand what it is, and will start asking for more, or will file issues regardless of what we write in the documentation.
> 
> If Biome starts supporting SCSS files, where's the line "this is supported, and this isn't?".
> 
> 

What about a plug-in?

**Comment by ThiefMaster** [comment]
I'd also like to see at least partial SCSS formatting. Fine to not format files using obscure syntax (e.g. indentation-based Sass) or more complex features, but if the vast majority of files that just use basic things which look like valid CSS anyway, and maybe simple things like `@mixin` syntax, were formatted, it'd already be a huge plus.

**Comment by ws-rob** [comment]
This would be a fantastic feature. It's the only reason we're still hanging on to Prettier on projects with SCSS. I appreciate @ematipico's point about where to draw the line, but SASS is still widely used, and has first-party support in cutting edge tools like Vite.

**Reply by ematipico** [reply]
I know it will, don't get me wrong. But we're just a few people, and we direct our contributions towards what we like most! 

I focused my time towards HTML and such because I also get something in return, and I need it.

Again, people can consider contributing and landing the feature. We didn't say no, we just don't have the capacity

**Reply by pooya-badiee** [reply]
@ematipico Just wanted to add my appreciation, Biome is a lot of work, and it already brings so much value even without SCSS support.  
Thank you to the whole Biome team!

**Reply by ws-rob** [reply]
@ematipico Your work is appreciated. I'm using Biome in some JS-only projects and it is amazing.

**Comment by XhstormR** [comment]
+1

**Comment by MrBns** [comment]
Is this possible to get `.SCSS` support this year ? :-)

**Comment by denbezrukov** [comment]
I've created an umbrella ticket to track the SCSS implementation.
https://github.com/biomejs/biome/issues/8732

**Comment by elliottmangham** [comment]
+1 would love this
